woocommerce/wc-admin
# Update payment gateway suggestions semantics to be more consistent

Fixes woocommerce/woocommerce-admin#6966 

* Removes all cases of `methods` in favor of `gateways`.
* Updates `recommendations` to `suggestions` for all suggested gateways.
* Uses `paymentGateway` only for actual installed payment gateways.
* Replaces `key` with `id` to match the `$gateway->id`.
* Renames the slots to match component names.
* Renames the feature to `PaymentGatewaySuggestions`.

### Detailed test instructions:

1.  Install some of the third party plugin gateways that include setup methods.  woocommerce/woocommerce#32258
2. Generally smoke test the payment suggestions list, making sure things work as expected.
3. Make sure slot fill and required settings still work as expected.
